pkg/controller/deployment_inplace.go

```diff
@@ -273,6 +274,11 @@ func (dc *controller) reconcileOldMachineSetsInPlace(ctx context.Context, allMac
 		return true, nil
 	}
 
+	// Manual inplace update
+	if deployment.Spec.Strategy.InPlaceUpdate.OrchestrationType == v1alpha1.OrchestrationTypeManual {
```

Comment:
For manual orchestration type, our understanding was that `MaxSurge` cannot be set as it would no longer be manual. In addition, MCM does not orchestrate in manual mode which means even if `MaxSurge` is set to any value > 0 then MCM will not be able to honor it. 

So the question is can we move this check above 266 line number where you check:
```go
if newMachineSet.Spec.Replicas == deployment.Spec.Replicas {
}
```

Reply:
I would prefer to keep it here, just to save any manual intervention issue.